providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/process/targeting/Operator.java

```diff
@@ -53,7 +59,10 @@ static class FlagProperties {
             if (from instanceof Map) {
                 Map<?, ?> dataMap = (Map<?, ?>) from;
 
-                Object flagKey = dataMap.get(FLAG_KEY);
+                Object flagKey = Optional.ofNullable(dataMap.get(FLAGD_PROPS_KEY))
+                    .filter(flagdProps -> flagdProps instanceof Map)
+                    .map(flagdProps -> ((Map<?, ?>)flagdProps).get(FLAG_KEY))
+                    .orElse(null);
```

Comment:
@Kavindu-Dodan there's not a lot of Optional usage in this repo, but the alternative was really ugly.